modules/renter/download.go

```diff
@@ -0,0 +1,150 @@
+package renter
+
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+// A Download is a download that has been queued by the renter.
+type Download struct {
+	completed   bool
+	destination string
+	nickname    string
+
+	renter *Renter
+}
```

Comment:
doesn't it? It satisfies a modules interface. Can you return unexported types as an interface?